### PR TITLE
fix WebGL context limit warnings by forcibly terminating the context

### PR DESCRIPTION
**Pull request by kand193**

### CODE REVIEW

File: src/transformers/BackgroundTransformer.ts [hunk 1]
```diff
@@ -60,6 +60,12 @@ export default class BackgroundProcessor extends VideoTransformer {
 
   async destroy() {
     await super.destroy();
+
+    const categoryMaskCanvas = this.segmentationResults?.categoryMask?.canvas;
+    if (categoryMaskCanvas) {
+      const gl = (categoryMaskCanvas.getContext('webgl2') ?? categoryMaskCanvas.getContext('webgl')) as WebGLRenderingContext;
+      gl?.getExtension('WEBGL_lose_context')?.loseContext();
+    }
```
Comment (lukasIO) [inline src/transformers/BackgroundTransformer.ts:68]: maybe move this whole block *after*  `this.imageSegmenter.close()`?

**Comment by wjaykim** [comment]
In my opinion, releasing resources must be handled by mediapipe side.
As an alternative, maybe passing shared canvas to[ ImageSegmenterOptions.canvas](https://github.com/google/mediapipe/blob/e8bf46f9fae1a6bba7f52899780f414a45115ca1/mediapipe/tasks/web/vision/core/vision_task_options.d.ts#L34) prevents creating new canvas and webgl contexts every time?

**Comment by lukasIO** [comment]
totally agree, that this is just a workaround for a bug in mediapipe! 

for the `canvas` option, I don't think it will work, the JS docs say
> The task will initialize a WebGL context and throw an error if
>   * this fails (e.g. if you have already initialized a different type of
>   * context).

which sounds a lot like they would still create a new context and if that isn't cleared, then throw an error...

**Comment by wjaykim** [comment]
I think the doc is saying like that because in MDN docs:

> Later calls to this method on the same canvas element, with the same contextType argument, will always return the same drawing context instance as was returned the first time the method was invoked. It is not possible to get a different drawing context object on a given canvas element.

Since we're not going to get context ourside but mediapipe will, they will get same type of context and won't produce error.

**Comment by lukasIO** [comment]
That makes sense! 
the segmenter options are exposed anyways, so the question would be if clearing (and passing) that canvas should be part of this library or if users of the library are in charge of handling it themselves. 
I think as long as the canvas option is exposed to users, then users would have to handle it. 
What do you think?

**Comment by wjaykim** [comment]
> I think as long as the canvas option is exposed to users, then users would have to handle it.

Same for me.
But for now, only `ImageSegmenterOptions.baseOption` can be customized by user. We will have to make users be able to customize entire `ImageSegmenterOptions`.

**Pull request closed without merging**